src/dbally/nl_responder/nl_responder.py

```diff
@@ -5,32 +5,35 @@
 
 from dbally.audit.event_tracker import EventTracker
 from dbally.data_models.execution_result import ViewExecutionResult
-from dbally.data_models.prompts.iql_explainer_prompt_template import (
-    IQLExplainerPromptTemplate,
-    default_iql_explainer_template,
-)
 from dbally.data_models.prompts.nl_responder_prompt_template import (
     NLResponderPromptTemplate,
     default_nl_responder_template,
 )
+from dbally.data_models.prompts.query_explainer_prompt_template import (
+    QueryExplainerPromptTemplate,
+    default_query_explainer_template,
+)
 from dbally.llm_client.base import LLMClient
 from dbally.nl_responder.token_counters import count_tokens_for_huggingface, count_tokens_for_openai
 
 
 class NLResponder:
     """Class used to generate natural language response from the database output."""
 
+    # Keys used to extract the query from the context (ordered by priority)
+    QUERY_KEYS = ["iql", "sql", "query"]
```

Comment:
What about API response? Maybe we should think about something generic that can be added to the context?

For now context basically can take a form of anything. Maybe we should think about Context dataclass or something? Context content seems to be consistent w.r.t the view type.

Reply:
What do you mean by "the API response"? The actual data? This is specifically a prompt fot the case when there is to much data to fit in the prompt, there is a separate prompt that includes the data (which is used whenever possible).

In general, I think there is a lot that can be improved when it comes to NL Responder with a lot of data but I think it would be best to create a separate ticket about researching possible solutions.

Reply:
Agree that separate ticket should be created. Besides this, everything clear.